Applications/Spire/Include/Spire/KeyBindings/KeyBindings.hpp

```diff
@@ -34,43 +39,50 @@ namespace Details {
   class KeyBindings {
     public:
 
+      /*! \class Tag
+          \brief Represents a tag with an optional value that can be passed to
+                 an order action.
+      */
+      class Tag {
+        public:
+
+          //! The type of a variant used to store all possible tag values.
+          using Type = typename Details::AddOptionalToVariant<
```

Comment:
Very ugly I know, but I could not come up with a better way to add types to the existing variant as I cannot handle two variadic sequences at a time. Let me know if you can think of a more elegant solution.